src/Catalog/Helpers/DeletionAuditEntry.cs

```diff
@@ -70,29 +70,44 @@ private DeletionAuditEntry(Uri uri, string contents)
             InitValues();
         }
 
+        [JsonConstructor]
+        public DeletionAuditEntry(Uri uri, JObject record, string id, string version, DateTime? timestamp)
```

Comment:
Can this be consistent with CatalogIndexEntry? - ctor without args and public setters; or ctor with args and private setters?